utilities/ordered-set.metta

```diff
@@ -0,0 +1,88 @@
+;; data structure for exemplars -- aka scored trees -- Xmplr
+;; deme ID
+(: demeID Type)
+(: mkDemeID (-> Number demeID))
+
+;; composite_score
+(: cscore Type)
+(: getCscore (-> tree cscore))
+        
+;; composite score components
+(: scoreT Type)
+(: cpxyT Type)
+
+(: mkScoreT (-> Number scoreT))
+(: mkCpxy (-> Number cpxyT))
+
+;; composite score is composed of a raw score, comlexity score, a complexity penalty and a uniformity penalty score
+(: mkCscore (-> scoreT cpxyT scoreT scoreT cscore))
+
+;; behavioral_score
+(: bscore Type)
+(: mkBscore (-> Number bscore))
+
+(: Xmplr Type)
+
+(: mkXmplr (-> tree demeID cscore bscore Xmplr))
+
+;; XXX - SUGGESTION -- for the exemplars, I have tried to follow the type definitions as they are defined in C++ MOSES
```

Comment:
The type definitions were actually needed to help us deferentiate what type of each number is. This can help prevent accidental variable passing with in our logic as we will get additional help from the type checker.